addons/pos_hr/models/account_bank_statement.py

```diff
@@ -6,3 +6,9 @@ class AccountBankStatementLine(models.Model):
     _inherit = 'account.bank.statement.line'
 
     employee_id = fields.Many2one('hr.employee', string="Employee", help="The employee who made the cash move.")
+
+    def _get_cashier_name(self):
+        self.ensure_one()
+        if self.employee_id:
+            return self.employee_id.name
+        return super()._get_cashier_name()
```

Comment:
get rid of this method, and just focus on using the `partner_id` field of the `'account.bank.statement.line'` model. Besides, I already see that you're passing the res.partner linked to the hr.employee.